pkg/archive/archive_unix.go

```diff
@@ -57,6 +58,10 @@ func minor(device uint64) uint64 {
 // handleTarTypeBlockCharFifo is an OS-specific helper function used by
 // createTarFile to handle the following types of header: Block; Char; Fifo
 func handleTarTypeBlockCharFifo(hdr *tar.Header, path string) error {
+	if shared.RunningInUserNS() {
```

Comment:
On Fri, Jan 08, 2016 at 09:27:46AM -0800, Phil Estes wrote:
> > @@ -57,6 +58,10 @@ func minor(device uint64) uint64 {
> >  // handleTarTypeBlockCharFifo is an OS-specific helper function used by
> >  // createTarFile to handle the following types of header: Block; Char; Fifo
> >  func handleTarTypeBlockCharFifo(hdr *tar.Header, path string) error {
> > +	if shared.RunningInUserNS() {
> 
> True. I went and looked at the function; it reads `/proc/self/uid_map` with some validation that maps are actually in use. As much as code reuse is good, vendoring in that tree seems potentially overkill. This may be a useful as a utility fn across the engine--what do you think of adding the same function to Docker @hallyn?

100% agreed - since I'm not particularly familiar with the coding style I
was hoping someone who was would reply with a counter-PR  and avoid any
bike-shedding :)

But I'll go ahead and take a stab in a few hours